### PR TITLE
diagnostics timestamps validation improvements

### PR DESCRIPTION
**Pull request by PhilipDeegan**
a few tests to add to test_validation.py

### CODE REVIEW

**Comment by PhilipDeegan** [comment]
need to handle the diags_dir trailing / messing things up

File: pyphare/pyphare/pharein/simulation.py [hunk 1]
```diff
@@ -395,6 +395,10 @@ def check_restart_options(**kwargs):
         if "mode" not in restart_options:
             raise ValueError (f"Restart mode not set, valid modes are {valid_modes}")
 
+        if "dir" in restart_options:
+            while restart_options["dir"][-1] == "/": # samrai can't handle this
```
Comment (PhilipDeegan) [inline pyphare/pyphare/pharein/simulation.py:399]: maybe should be 

```
while len(restart_options["dir"]) > 1 restart_options["dir"][-1] == "/
```

?

Comment (PhilipDeegan) [reply pyphare/pyphare/pharein/simulation.py:399]: actually 

```
while "//" in restart_options["dir"]: restart_options["dir"] = restart_options["dir"].replace("//", "/")
```

Comment (PhilipDeegan) [reply pyphare/pyphare/pharein/simulation.py:399]: or maybe just raise and say it's invalid

Comment (nicolasaunai) [reply pyphare/pyphare/pharein/simulation.py:399]: "aaa/bbb///".rstrip("/")

Comment (nicolasaunai) [reply pyphare/pyphare/pharein/simulation.py:399]: IMO we should also do other checks such as permissions (https://www.tutorialspoint.com/How-to-check-the-permissions-of-a-directory-using-Python), other things?

Comment (PhilipDeegan) [reply pyphare/pyphare/pharein/simulation.py:399]: check there's some space left on the disk maybe

Comment (nicolasaunai) [reply pyphare/pyphare/pharein/simulation.py:399]: probably not straightforward with the quotas

File: pyphare/pyphare/pharein/diagnostics.py [hunk 2]
```diff
@@ -145,7 +143,7 @@ def to_dict(self):
 # ------------------------------------------------------------------------------
 
 def population_in_model(population):
-    return population in [p for p in global_vars.sim.model.populations] + ["all","ions"]
+    return population in [p for p in global_vars.sim.model.populations]
```
Comment (nicolasaunai) [inline pyphare/pyphare/pharein/diagnostics.py:146]: don't remember why we dropped those. I think it was causing a bug somewhere?

**Review by nicolasaunai — APPROVED** [review]

File: pyphare/pyphare/pharein/diagnostics.py (hunk 2 above)
Comment (PhilipDeegan) [reply pyphare/pyphare/pharein/diagnostics.py:146]: yeah, I think there was something

**Pull request merged**